src/Compilers/CSharp/Portable/Declarations/DeclarationTable.cs

```diff
@@ -148,9 +148,12 @@ internal RootNamespaceLocationComparer(CSharpCompilation compilation)
                 _compilation = compilation;
             }
 
+            [PerformanceSensitive(
+                "https://github.com/dotnet/roslyn/issues/23582",
+                Constraint = "Avoid " + nameof(SingleNamespaceOrTypeDeclaration.Location) + " since it has a costly allocation on this fast path.")]
```

Comment:
Why an attribute rather than a comment?

Reply:
➡️ The attribute is capable of driving analyzers to prevent regressions. Aside from the initial benefit of enforcing that the first parameter is a valid URI, one example that I figured out already is dotnet/roslyn-analyzers#1439. I haven't figured out a general form for "prevent access to [some specific item]", so I used the general purpose `Constraint` property here for now.